handlers/messagebird/messagebird.go

```diff
@@ -0,0 +1,309 @@
+package messagebird
+
+/*
+ * Handler for MessageBird
+ */
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"strconv"
+
+	"fmt"
+
+	"net/http"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	smsURL                    = "https://rest.messagebird.com/messages"
+	mmsURL                    = "https://rest.messagebird.com/mms"
+	signatureHeader           = "Messagebird-Signature-Jwt"
+	maxRequestBodyBytes int64 = 1024 * 1024
+	// error code messagebird returns when a contact has sent "stop"
+	errorStopped = 103
+)
+
+type Message struct {
+	Recipients []string `json:"recipients"`
+	Reference  string   `json:"reference,omitempty"`
+	Originator string   `json:"originator"`
+	Subject    string   `json:"subject,omitempty"`
+	Body       string   `json:"body,omitempty"`
+	MediaURLs  []string `json:"mediaUrls,omitempty"`
+}
+
+type ReceivedStatus struct {
+	ID              string
+	Reference       string
+	Recipient       string
+	Status          string
+	StatusReason    string
+	StatusDatetime  time.Time
+	StatusErrorCode int
+}
+
+var statusMapping = map[string]courier.MsgStatusValue{
+	"scheduled":       courier.MsgSent,
+	"delivery_failed": courier.MsgFailed,
+	"sent":            courier.MsgSent,
+	"buffered":        courier.MsgSent,
+	"delivered":       courier.MsgDelivered,
+	"expired":         courier.MsgFailed,
+}
+
+type ReceivedMessage struct {
+	ID              string    `json:"id"`
+	Recipient       string    `json:"recipient"`
+	Originator      string    `json:"originator"`
+	Body            string    `json:"body"`
+	CreatedDatetime time.Time `json:"createdDatetime"`
+	MediaURLs       []string  `json:"mediaUrls"`
+	MMS             bool      `json:"mms"`
+}
+
+func init() {
+	courier.RegisterHandler(newHandler("MBD", "Messagebird", true))
+}
+
+type handler struct {
+	handlers.BaseHandler
+	validateSignatures bool
+}
+
+func newHandler(channelType courier.ChannelType, name string, validateSignatures bool) courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("MBD"), "Messagebird"), validateSignatures}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	s.AddHandlerRoute(h, http.MethodPost, "receive", courier.ChannelLogTypeMsgReceive, handlers.JSONPayload(h, h.receiveMessage))
+	s.AddHandlerRoute(h, http.MethodGet, "status", courier.ChannelLogTypeMsgStatus, h.receiveStatus)
+	return nil
+}
+
+func (h *handler) receiveStatus(ctx context.Context, channel courier.Channel, w http.ResponseWriter, r *http.Request, clog *courier.ChannelLog) ([]courier.Event, error) {
```

Comment:
this can use `handlers.JSONPayload` too no?

Reply:
yup. good call